### PR TITLE
Add implicit_account_creation_fee option for non-token accounts

### PR DESCRIPTION
**Pull request by mrmr1993**
This PR adds a `implicit_account_creation_fee` field to parties, which can be used (on non-token parties) to restore the old behaviour of optionally subtracting the creation fee from the transferred amount.

This PR builds upon #10836.

Checklist:

- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them

### CODE REVIEW

**Review by mitschabaude — APPROVED** [review]
LGTM!

File: src/lib/transaction_logic/parties_logic.ml [hunk 1]
```diff
@@ -1405,22 +1409,43 @@ module Make (Inputs : Inputs_intf) = struct
...
+      let local_delta_for_creation, `Overflow creation_overflow =
+        let open Amount.Signed in
+        add_flagged local_delta (of_unsigned account_creation_fee)
+      in
+      let pay_creation_fee =
+        Bool.(account_is_new &&& implicit_account_creation_fee)
+      in
+      let creation_overflow = Bool.(pay_creation_fee &&& creation_overflow) in
+      let local_delta =
+        Amount.Signed.if_ pay_creation_fee ~then_:local_delta_for_creation
```
Comment (deepthiskumar) [inline src/lib/transaction_logic/parties_logic.ml:1432]: In the case of default-token accounts, are we deducting account creation fee for an account twice? 
And, for token accounts what's the rationale for burning custom tokens?

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Review by ghost-not-in-the-shell — APPROVED** [review]

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Pull request merged**